src/orange/g4org/SolidConverter.cc

```diff
@@ -609,55 +609,56 @@ auto SolidConverter::trap(arg_type solid_base) -> result_type
 {
     auto const& solid = dynamic_cast<G4Trap const&>(solid_base);
 
-    real_type sin_phi{}, cos_phi{}, tan_theta{};
+    real_type tan_theta{};
+    Turn phi{};
 #if G4VERSION_NUMBER < 1100
     // Geant4 10.7 and earlier - axis = (sinth.cosphi, sinth.sinphi, costh)
     // Note: both sinth,costh >= 0 since theta is in [0, pi/2] for a G4Trap
     auto axis = solid.GetSymAxis();
-    auto sin_theta = std::sqrt(real_type(1.0) - axis.z() * axis.z());
+    auto sin_theta = std::sqrt(1.0 - axis.x() * axis.x() + axis.y() * axis.y());
```

Comment:
Might want to put a `max` in here in case of floating point roundoff, and use `ipow`
```suggestion
    auto sin_theta = std::sqrt(max(0.0, 1.0 - ipow<2>(axis.x()) + ipow<2>(axis.y()));
```

Reply:
Done, with a correction: should be sqrt(1.0 - ipow<2>(axis.z())).